### PR TITLE
HubConfig API changes

### PR DESCRIPTION
**Pull request by akshaysonvane**
Once Spring generates the hubConfig, the user can set properties using setters and call refreshProject which will take care of loading all properties and wiring clients based on these properties.

### CODE REVIEW

**Comment by akshaysonvane** [comment]
Closing PR for now. Need to make a few more changes.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubConfigImpl.java [hunk 1]
```diff
@@ -78,28 +76,23 @@
...
-    @Autowired Versions versions;
+    @Autowired
+    FlowManagerImpl flowManager;
+    @Autowired
+    DataHubImpl dataHub;
+    @Autowired
+    Versions versions;
+
 
-    @Value("${mlHost}")
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubConfigImpl.java]: I'm nervous removing these - what problem does removing these annotations fix?

**Comment by akshaysonvane** [comment]
Brought in @srinathgit changes from #1660

File: marklogic-data-hub/src/main/java/com/marklogic/hub/HubConfig.java [hunk 2]
```diff
@@ -541,4 +539,12 @@
      * @return
      */
     HubConfig withPropertiesFromEnvironment(String environment);
+
+    /**
+     * Loads HubConfig object with values from gradle.properties (optionally overriden with
+     * gradle-(environment).properties). This method must be called every time after Spring creates
+     * HubConfig object and the project is initialized with createProject(String projectDirString)
+     * method
+     */
```
Comment (srinathgit) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/HubConfig.java]: @akshaysonvane , should the doc include something to the effect of "make all changes wanted  to "hubConfig" using set methods and then call refreshProject" ?

Comment (akshaysonvane) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/HubConfig.java]: Updated description 9d653dc

**Pull request merged**